### PR TITLE
Last batch of javaharness lint/errorprone/style fixes

### PR DESCRIPTION
**Pull request by cromwellian**

### CODE REVIEW

**Comment by cromwellian** [comment]
After these changes, should be tricorder free of errors.

File: src/javaharness/java/arcs/api/SlotProxy.java [hunk 1]
```diff
@@ -33,6 +33,6 @@ void render(String content) {
     contentJson.forEach(requestedContentTypes::remove);
     // Slot is considered rendered, if a non-empty content was sent and all requested content types
     // were fullfilled.
-    boolean isRendered = requestedContentTypes.size() == 0 && contentJson.keys().size() == 0;
+    //    boolean isRendered = requestedContentTypes.size() == 0 && contentJson.keys().size() == 0;
   }
```
Comment (shaper) [inline src/javaharness/java/arcs/api/SlotProxy.java]: This seems like a bug in our Java implementation, unless it's ancillary / only important while doing dev on the Java PEC?

File: src/javaharness/java/arcs/crdt/CollectionOperation.java [hunk 2]
```diff
@@ -23,8 +23,6 @@ public CollectionOperation(Type type, T t, VersionMap clock, String actor) {
       case REMOVE:
         this.removed = Optional.of(t);
         break;
-      default:
```
Comment (shaper) [inline src/javaharness/java/arcs/crdt/CollectionOperation.java]: Maybe we should at least log something? Was the error about throwing inside of a ctor, or around use of `default`?

File: src/javaharness/.bazelrc [hunk 3]
```diff
@@ -13,5 +13,5 @@ build --watchfs
 build --spawn_strategy=local
 build --strategy=J2cl=worker
 build --strategy=Closure=worker
-build --experimental_persistent_javac
+#build --experimental_persistent_javac
```
Comment (shaper) [inline src/javaharness/.bazelrc]: Why comment out? Maybe add a comment explaining?

File: src/javaharness/java/arcs/crdt/CollectionOperation.java (hunk 2 above)
Comment (cromwellian) [reply src/javaharness/java/arcs/crdt/CollectionOperation.java]: According to error prone, all switch enums are handled, so the fallthrough is impossible

File: src/javaharness/java/arcs/api/SlotProxy.java (hunk 1 above)
Comment (cromwellian) [reply src/javaharness/java/arcs/api/SlotProxy.java]: It's considered deadcode by errorprone, this line doesn't do anything, I dunno why its there, so I commented it out.

File: src/javaharness/.bazelrc (hunk 3 above)
Comment (cromwellian) [reply src/javaharness/.bazelrc]: reverting, i was trying to debug a compiler crash

File: src/javaharness/java/arcs/api/SlotProxy.java (hunk 1 above)
Comment (cromwellian) [reply src/javaharness/java/arcs/api/SlotProxy.java]: @mmandlis should answer this one

Comment (mmandlis) [reply src/javaharness/java/arcs/api/SlotProxy.java]: it was `this.isRendered = ` until https://github.com/PolymerLabs/arcs/pull/3468. 
`this` is not needed, but the code should update a data member, not a local variable.

**Pull request merged**